ipyparallel/cluster/shellcmd.py

```diff
@@ -0,0 +1,860 @@
+            self._log("ShellCommandReceive instance created")
+
+    def __del__(self):
+        if self.log:
+            self._log("ShellCommandReceive instance deleted")
+
+    def _linux_quote(self, p):
+        if "'" in p:
+            return '"' + p + '"'
+        else:
+            return p
+
+    def cmd_start(self, start_cmd, env=None, output_file=None):
+        if env:
+            # add provided entries to environment
+            if isinstance(env, str):
+                if env[0] == '"' and env[-1] == '"':
+                    env = env.strip('"')  # occurs under windows cmd
+                if "\\'" in env:
+                    env = env.replace("\\'", "'")  # replace quoted
+                env_dict = eval(env)
+                assert isinstance(env_dict, dict) is True
+                env = env_dict
+
+            assert isinstance(env, dict) is True  # make sure that env is a dictionary
+
+            # update environment
+            for key, value in env.items():
+                if value is not None and value != '':
+                    if isinstance(value, dict):
+                        os.environ[key] = json.dumps(value)
+                    else:
+                        os.environ[key] = str(value)
+                else:
+                    # unset entry if needed
+                    if key in os.environ:
+                        del os.environ[key]
+
+        if isinstance(start_cmd, str):
+            start_cmd = [start_cmd]
+
+        if self.platform == Platform.Windows:
+            # under windows we need to remove embracing double quotes
+            for idx, p in enumerate(start_cmd):
+                if not isinstance(p, str):
+                    start_cmd[idx] = str(p)
+                    continue
+                if p[0] == '"' and p[-1] == '"':
+                    start_cmd[idx] = p.strip('"')
+
+        self._log(f"start_cmd={start_cmd}  (use_break_way={self.use_break_way})")
+
+        if self.platform == Platform.Windows:
+            from subprocess import (
+                CREATE_BREAKAWAY_FROM_JOB,
+                CREATE_NEW_CONSOLE,
+                DEVNULL,
+                Popen,
+            )
+
+            flags = 0
+            if self.use_break_way != False:
+                flags |= CREATE_NEW_CONSOLE
+                flags |= CREATE_BREAKAWAY_FROM_JOB
+
+            pkwargs = {
+                'close_fds': True,  # close stdin/stdout/stderr on child
+                'creationflags': flags,
+            }
+            if output_file:
+                fo = open(output_file, "w")
+                pkwargs['stdout'] = fo
+                pkwargs['stderr'] = fo
+                pkwargs['stdin'] = DEVNULL
+
+            self._log(f"Popen(**pkwargs={pkwargs}")
+            p = Popen(start_cmd, **pkwargs)
+            self._log(f"pid={p.pid}")
+
+            print(f'__remote_pid={p.pid}__')
+            sys.stdout.flush()
+            if self.use_break_way == False:
```

Comment:
Let's avoid comparing bools with bools and then checking the comparison:

```suggestion
            if not self.use_breakaway:
```

Reply:
You are right! Fixed in commit https://github.com/ipython/ipyparallel/commit/0a0efb02be1da21c224f9e805159c13809902e4f